packages/editor/src/components/post-schedule/label.js

```diff
@@ -2,16 +2,23 @@
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { format, __experimentalGetSettings } from '@wordpress/date';
+
+import {
+	dateI18n,
+	getActualTimezone,
+	__experimentalGetSettings,
+	zonedTimeToUtc,
+} from '@wordpress/date';
+
 import { withSelect } from '@wordpress/data';
 
 export function PostScheduleLabel( { date, isFloating } ) {
 	const settings = __experimentalGetSettings();
 
 	return date && ! isFloating
-		? format(
+		? dateI18n(
 				`${ settings.formats.date } ${ settings.formats.time }`,
-				date
+				zonedTimeToUtc( date, getActualTimezone() )
```

Comment:
Simply formatting a date that comes in the WordPress timezone, and keeping in the WordPress seems like a very common. Developers should not need to call zonedTimeToUtc and getActualTimezone. It seems we don't have a function that formats a date that comes in the WordPress timezone (inexplicitly like the dates from WordPress API) and formats it keeping the WordPress timezone. On https://github.com/WordPress/gutenberg/pull/26276 I'm proposing a new function that does that (and should probably the function used in 99% of the cases where date formatting in WordPress is needed). What do you think about the new function, do you think we need something new or we can adapt what we have here?

Reply:
I think we should be able to move this operation (`zonedTimeToUtc( date, getActualTimezone() )`) into the `dateI18n` function under a parameter, or even make it the default way in which it runs, which would make `dateI18n` work as the one proposed on #26276. Either way, I think that using the current `timezone` parameter on `dateI18n`, and looking for timezone fragments of the timestamp can help us get there.

>  Developers should not need to call zonedTimeToUtc and getActualTimezone.

Agreed! We could either add the function proposed or do what I mention above instead of having developers call these methods manually.